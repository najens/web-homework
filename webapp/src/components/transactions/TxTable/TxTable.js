import React from 'react'
import { arrayOf, string, bool, number, shape, func } from 'prop-types'
import { css } from '@emotion/core'
import TxTableHeaderList from '../TxTableHeaderList'
import TxTableRowList from '../TxTableRowList'

export default function TxTable ({ data, handleOpenTxModal, refetchTransactions }) {
  return (
    <table css={tableStyle}>
      <tbody>
        <tr css={headerStyle}>
          <TxTableHeaderList refetchTransactions={refetchTransactions} />
        </tr>
        <TxTableRowList data={data} handleOpenTxModal={handleOpenTxModal} />
      </tbody>
    </table>
  )
}

TxTable.propTypes = {
  data: arrayOf(shape({
    amount: number,
    category: shape({
      id: string,
      name: string
    }),
    company: shape({
      id: string,
      name: string
    }),
    credit: bool,
    debit: bool,
    description: string,
    id: string,
    inserted_at: string,
    merchant: shape({
      id: string,
      name: string
    }),
    user: shape({
      id: string,
      firstName: string,
      lastName: string
    })
  })),
  handleOpenTxModal: func,
  refetchTransactions: func
}

const headerStyle = css`
  background: linear-gradient(180deg, #e4edf9, #62a2f9bf);
  font-size: 12px;
  font-weight: bold;
  overflow-x: scroll;
`
const tableStyle = css`
  border-collapse: collapse;
  width: 90vw;
`
