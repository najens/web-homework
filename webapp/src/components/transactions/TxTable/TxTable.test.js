import React from 'react'
import { create } from 'react-test-renderer'
import TxTable from './TxTable'

// Only testing render since this component has no business logic
describe('TxTable component', () => {
  it('renders correctly', () => {
    const data = [
      {
        __typename: 'Transaction',
        id: '782a1d8d-08dc-4ef8-bab0-d922038971a2',
        insertedAt: '2021-03-12T20:00:29',
        user: {
          __typename: 'User',
          firstName: 'Bobby',
          lastName: 'Smith'
        },
        category:{
          __typename: 'Category',
          id: '620c583-1307-4433-8900-8ad6d743f13d',
          name: 'Medical'
        },
        company:{
          __typename: 'Company',
          id: '520c583-1307-4433-8900-8ad6d743f13d',
          name: 'Test Company'
        },
        description: 'Hot Dogs',
        merchant: {
          __typename: 'Merchant',
          id: '420c583-1307-4433-8900-8ad6d743f13d',
          name: 'Taco Bell'
        },
        debit: true,
        credit: false,
        amount: 304
      }
    ]

    const testRenderer = create(
      <TxTable data={data} refetchTransactions={() => null} />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
