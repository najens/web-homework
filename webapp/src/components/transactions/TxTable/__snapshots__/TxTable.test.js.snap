// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxTable component renders correctly 1`] = `
.emotion-22 {
  border-collapse: collapse;
  width: 90vw;
}

.emotion-13 {
  background: linear-gradient(180deg,#e4edf9,#62a2f9bf);
  font-size: 12px;
  font-weight: bold;
  overflow-x: scroll;
}

.emotion-1 {
  border: 1px solid #b4c1ec;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 24px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0px 6px;
  width: 100%;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-21 {
  font-size: 12px;
  line-height: 2.5;
}

.emotion-21:hover {
  cursor: pointer;
}

.emotion-14 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

.emotion-20 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
  direction: rtl;
}

<table
  className="emotion-22"
>
  <tbody>
    <tr
      className="emotion-13"
    >
      <th
        className="emotion-1"
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <button
          className="emotion-0"
          onClick={[Function]}
        >
          <span>
            Date
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "fontSize": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </button>
      </th>
      <th
        className="emotion-1"
        style={
          Object {
            "width": "15%",
          }
        }
      >
        <button
          className="emotion-0"
          onClick={[Function]}
        >
          <span>
            User
          </span>
        </button>
      </th>
      <th
        className="emotion-1"
        style={
          Object {
            "width": "25%",
          }
        }
      >
        <button
          className="emotion-0"
          onClick={[Function]}
        >
          <span>
            Description
          </span>
        </button>
      </th>
      <th
        className="emotion-1"
        style={
          Object {
            "width": "15%",
          }
        }
      >
        <button
          className="emotion-0"
          onClick={[Function]}
        >
          <span>
            Merchant
          </span>
        </button>
      </th>
      <th
        className="emotion-1"
        style={
          Object {
            "width": "15%",
          }
        }
      >
        <button
          className="emotion-0"
          onClick={[Function]}
        >
          <span>
            Category
          </span>
        </button>
      </th>
      <th
        className="emotion-1"
        style={
          Object {
            "width": "8%",
          }
        }
      />
      <th
        className="emotion-1"
        style={
          Object {
            "width": "12%",
          }
        }
      >
        <button
          className="emotion-0"
          onClick={[Function]}
        >
          <span>
            Amount
          </span>
        </button>
      </th>
    </tr>
    <tr
      className="emotion-21"
      data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2"
      onClick={[Function]}
    >
      <td
        className="emotion-14"
        data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-date"
      >
        Mar 12
      </td>
      <td
        className="emotion-14"
        data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-userName"
      >
        Bobby Smith
      </td>
      <td
        className="emotion-14"
        data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-description"
      >
        Hot Dogs
      </td>
      <td
        className="emotion-14"
        data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-merchantName"
      >
        Taco Bell
      </td>
      <td
        className="emotion-14"
        data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-category"
      >
        Medical
      </td>
      <td
        className="emotion-14"
        data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-filler"
      >
        
      </td>
      <td
        className="emotion-20"
        data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-amount"
      >
        3.04 $
      </td>
    </tr>
  </tbody>
</table>
`;
